core/src/main/java/com/ttbmp/cinehub/core/usecase/buyticket/BuyTicketUseCaseController.java

```diff
@@ -0,0 +1,272 @@
+
+
+    public BuyTicketUseCaseController(PaymentService paymentService,
+                                      EmailService emailService,
+                                      BuyTicketPresenter buyTicketPresenter,
+                                      MovieApiService movieApiService,
+                                      MovieRepository movieRepository,
+                                      CinemaRepository cinemaRepository,
+                                      TimeRepository timeRepository,
+                                      SeatRepository seatRepository,
+                                      AuthenticationService authenticationService,
+                                      HallRepository hallRepository,
+                                      ProjectionRepository projectionRepository,
+                                      UserRepository userRepository) {
+        this.paymentService = paymentService;
+        this.emailService = emailService;
+        this.buyTicketPresenter = buyTicketPresenter;
+        this.movieApiService = movieApiService;
+        this.movieRepository = movieRepository;
+        this.cinemaRepository = cinemaRepository;
+        this.timeRepository = timeRepository;
+        this.seatRepository = seatRepository;
+        this.authenticationService = authenticationService;
+        this.hallRepository = hallRepository;
+        this.projectionRepository = projectionRepository;
+        this.userRepository = userRepository;
+
+    }
+
+
+
+    @Override
+    public boolean pay(PayRequest request) {
+        try {
+            Request.validate(request);
+            int userId = authenticationService.sigIn();
+            User user = userRepository.getUser(userId).getValue();
+            TicketAbstract ticketAbstract = TicketDataMapper.mapToEntity(request.getTicket());
+            Projection projection = ProjectionDataMapper.mapToEntity(request.getProjection());
+            Integer index = request.getIndex();
+            if (paymentService.pay(new PayServiceRequest(user.getEmail(), user.getName(), user.getCard().getNumber(), ticketAbstract.getPrice()))) {
+                ticketAbstract.setState(true);
+                projection.addTicket(ticketAbstract);
+                projection.getHall().getSeatList().get(index).setState(false);
+                user.addTicket(ticketAbstract);
+                emailService.sendMail(new EmailServiceRequest(user.getEmail(), "Payment receipt"));
+                return true;
+            }
+            buyTicketPresenter.presentErrorByStripe(paymentService.getError());
+            return false;
+        } catch (Request.NullRequestException e) {
+            buyTicketPresenter.presentPayNullRequest();
+            return false;
+        } catch (Request.InvalidRequestException e) {
+            buyTicketPresenter.presentInvalidPay(request);
+            return false;
+        }
+    }
+
+    @Override
+    public void getListMovie(String localDate) {
```

Comment:
There is no validation check for the `localDate` parameter, instead of mixing error logic with business logic it would be better to use a request object instead.

Reply:
I remember that the professor told me that if a string is passed as the only parameter, it was not necessary to insert a bean, therefore a request